### PR TITLE
9 reward service implement fitness

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 1]
```diff
@@ -23,8 +45,71 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
+
+        // Check if the repetition date is today or in the past
+        boolean isDueForRepetition = repetitionDate.isBefore(today) || repetitionDate.isEqual(today);
+
+        return isDueForRepetition;
+    }
+    private int calculateDaysOverdue(Content content) {
+
+        OffsetDateTime today = OffsetDateTime.now();
+        OffsetDateTime repetitionDate = content.getMetadata().getSuggestedDate();
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:94]: the suggested date is not the repetition date! the repetition date is "content.getUserData().getLastLearnDate()" (or similar)

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 2]
```diff
@@ -23,8 +45,71 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
+            }
+        }
+        return Math.min(MAX_DECREASE_PER_DAY, fitnessDecrease);
+    }
+
+    private boolean isDueForRepetition(Content content) {
+        // Implement the logic to determine if the content is due for repetition
+        // Return true if it is due, false otherwise
+        OffsetDateTime today = OffsetDateTime.now();
+        OffsetDateTime repetitionDate = content.getMetadata().getSuggestedDate();
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:84]: See comment below

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 3]
```diff
@@ -23,8 +45,71 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
+
+        // Calculate the number of days between the current date and the repetition date
+        long daysBetween = ChronoUnit.DAYS.between(repetitionDate, today);
+
+        // If the content is due for repetition today, set daysOverdue to 1
+        int daysOverdue = daysBetween > 0 ? (int) daysBetween : 1;
+
+        return daysOverdue;
+    }
+    private double calculateCorrectness(UserProgressLogEvent event) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:104]: i would rename this slightly to `calculateCorrectnessModifier`

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 4]
```diff
@@ -23,8 +45,71 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
     }
+
+    private double calculateFitnessDecrease(List<Content> contents, UserProgressLogEvent event) {
+        double fitnessDecrease = 0.0;
+
+        for (Content content : contents) {
+            if (isDueForRepetition(content)) {
+                int daysOverdue = calculateDaysOverdue(content);
+                double correctness = calculateCorrectness(event);
+                double decreasePerDay = 1 + (2 * daysOverdue * (1 - Math.pow(correctness, 2)));
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:73]: you are already doing the power in the `calculateCorrectness` method below, right? so here it would be done again

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 5]
```diff
@@ -2,19 +2,41 @@
...
 
-import java.util.List;
+import java.time.OffsetDateTime;
 
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
 @Component
 public class FitnessScoreCalculator implements ScoreCalculator {
+    private static final double MAX_DECREASE_PER_DAY = 0.2;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:18]: fitness is between 0 and 100, so I think this should be 20, not 0.2

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 6]
```diff
@@ -2,19 +2,41 @@
...
     @Override
     public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores, List<Content> contents) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        UserProgressLogEvent event = UserProgressLogEvent.builder().build();
+
+        double fitnessDecrease = calculateFitnessDecrease(contents,event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
+        int IntnewFitnessScore= (int) Math.round(newFitnessScore);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:28]: variable should start with lower case letter

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 7]
```diff
@@ -23,8 +45,71 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        double fitnessDecrease = calculateFitnessDecrease(contents,event);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:48]: this method is not doing the right thing. When the user worked on a content, it should decrease its fitness. I would suggest you have a look at how it is done in the health calculator.

**Review by PaulBredl — CHANGES_REQUESTED** [review]

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 3 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:104]: done

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 5 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:18]: done

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 2 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:84]: I did to get it here,,,,

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 7 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:48]: done

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 4 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:73]: done

**Comment by Keyshal-211** [comment]
In updated code, I rewrite the code what you @PaulBredl suggested me to change. Please review it and let me know, if something wrong

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 8]
```diff
@@ -23,8 +45,76 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
+        // Decrease the fitness score
+        int fitnessDecreaseValue = fitnessScoreBefore.getValue() - newFitnessScoreInt;
+        int updatedFitnessScore = fitnessScoreBefore.getValue() - fitnessDecreaseValue;
+
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+                .date(OffsetDateTime.now())
+                .difference(fitnessDecreaseValue)
+                .oldValue(fitnessScoreBefore.getValue())
+                .newValue(updatedFitnessScore)
+                .reason(RewardChangeReason.CONTENT_DONE)
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:61]: There is a specific reward change reason for content that was repeated CONTENT_REVIEWED or something

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 9]
```diff
@@ -23,8 +45,76 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
+                .associatedContentIds(Collections.emptyList())
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:62]: This should not be empty, the user should know which content it was, that increased his fitness

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 10]
```diff
@@ -2,19 +2,41 @@
...
+
     @Override
     public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores, List<Content> contents) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        UserProgressLogEvent event = UserProgressLogEvent.builder().build();
+
+        double fitnessDecrease = calculateFitnessDecrease(contents,event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:27]: This is not correct. Let's say you currently have a fitness of 100, and fitness descrease would be 20. Then this formula would return: 100 * (1 - 20) = 100 * -19 = -1900
But fitness is a value between 1 and 100. Why not just `fitnessScoreBefore.getValue() - fitnessDecrease`?

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 11]
```diff
@@ -2,19 +2,41 @@
...
+        int intnewFitnessScore= (int) Math.round(newFitnessScore);
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+               .date(OffsetDateTime.now())
+                .difference(fitnessScoreBefore.getValue() - intnewFitnessScore)
+                .oldValue(fitnessScoreBefore.getValue())
+                .newValue(intnewFitnessScore)
+                .reason(RewardChangeReason.CONTENT_DUE_FOR_LEARNING)
+                .associatedContentIds(Collections.emptyList())
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:35]: which contents are due for learning? This may not be empty

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 12]
```diff
@@ -2,19 +2,41 @@
...
 @Component
 public class FitnessScoreCalculator implements ScoreCalculator {
+    private static final double MAX_DECREASE_PER_DAY = 20;
+
     @Override
     public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores, List<Content> contents) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        UserProgressLogEvent event = UserProgressLogEvent.builder().build();
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:24]: why are you creating an event? There should be no event. If you need an event then your logic is not right

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 1 above)
Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:99]: No it's not?

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 13]
```diff
@@ -23,8 +45,76 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        double fitnessDecrease = calculateFitnessDecrease(contents, event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
+        int newFitnessScoreInt = (int) Math.round(newFitnessScore);
+
+        // Decrease the fitness score
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:52]: why does fitness decrease when content was learned? That is not the correct logic. The fitness should increase! I think you have to look at the formulas in the wiki again.

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 14]
```diff
@@ -23,8 +45,76 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
...
+        return Math.min(MAX_DECREASE_PER_DAY, fitnessDecrease);
+    }
+    private boolean isDueForRepetition(Content content) {
+        // Implement the logic to determine if the content is due for repetition
+        // Return true if it is due, false otherwise
+        OffsetDateTime today = OffsetDateTime.now();
+        OffsetDateTime repetitionDate = content.getUserProgressData().getLastLearnDate();
+
+        // Check if the repetition date is today or in the past
+        boolean isDueForRepetition = repetitionDate.isBefore(today) || repetitionDate.isEqual(today);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:92]: This is very good btw, I would have forgotten to check for equal 👍

**Review by PaulBredl — CHANGES_REQUESTED** [review]

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 13 above)
Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:52]: image

Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:52]: (there is a small error in the formula in the wiki, should be (100 - fitness) in the parenthesis)

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 8 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:61]: done

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 1 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:99]: done

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 2 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java]: done

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java (hunk 12 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:24]: removed it

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 15]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        OffsetDateTime today = OffsetDateTime.now();
-        double fitnessDecrease = calculateFitnessDecrease(contents, today);
-        double newFitnessScore = fitnessScoreBefore.getValue() - fitnessDecrease ;
+        double fitnessDecrease = calculateFitnessDecrease(contents);
+        double newFitnessScore = fitnessScoreBefore.getValue() - fitnessDecrease;
         int intNewFitnessScore = (int) Math.round(newFitnessScore);
 
+        if (fitnessDecrease == 0) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:30]: this is required to not create a log entry when the differene is zero

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 16]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
     @Override
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
-        OffsetDateTime today = OffsetDateTime.now();
-        double fitnessDecrease = calculateFitnessDecrease(contents, today);
-        double newFitnessScore = fitnessScoreBefore.getValue() - fitnessDecrease;
-        int intNewFitnessScore = (int) Math.round(newFitnessScore);
 
-        double correctnessBefore = calculateCorrectnessBefore(contents);
+        Content content = getContentOfEvent(contents, event);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:53]: you considered all contents in your calculation, only the content that was learned triggering the event should be considered!

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 17]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
 
-        double fitnessRegen = calculateFitnessRegeneration(newFitnessScore, contentsToRepeat, correctnessBefore, correctnessAfter);
+        double fitnessRegen = calculateFitnessRegeneration(fitnessScoreBefore.getValue(), contentsDueForReview.size(),
+                correctnessBefore, correctnessAfter);
 
-        // Cap the fitness regeneration at 1% per repetition
-        fitnessRegen = Math.min(fitnessRegen, 1);
+        if (fitnessRegen == 0
+            || !event.isSuccess()
+            || wasAlreadyLearnedToday(latestReview)
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:65]: Don't create lgo entries when the learning was not successful or it was already learned today.

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 18]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
+        ) {
+            // no change in fitness score or content was not reviewed successfully
+            // or content was already learned today
+            return fitnessScoreBefore;
+        }
 
-        double updatedFitnessScore = newFitnessScore + fitnessRegen * contentsToRepeat;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java]: no need to multiply anything

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 19]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
+        double updatedFitnessScore = fitnessScoreBefore.getValue() + fitnessRegen;
         int intUpdatedFitnessScore = (int) Math.round(updatedFitnessScore);
 
+        if (!isDueForRepetition(content)) {
+            // content was reviewed successfully but is not due for repetition
+            intUpdatedFitnessScore = fitnessScoreBefore.getValue() + 1;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:77]: special case (was described in wiki)

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 20]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
+        return lastReview.getTimestamp().truncatedTo(ChronoUnit.DAYS)
+                .equals(OffsetDateTime.now().truncatedTo(ChronoUnit.DAYS));
+    }
+
+    private double calculateFitnessDecrease(List<Content> contents) {
         double fitnessDecrease = 0.0;
 
         for (Content content : contents) {
-            if (isDueForRepetition(content)) {
+            if (isDueForRepetition(content) && isNotNew(content)) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java:115]: also important: contents that were never learned may not be considered for the fitness

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 21]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
 
-        System.out.println("Days Overdue: " + daysOverdue);
-
-        return daysOverdue;
+        // add one because at the repetition date the content is already overdue
+        return (int) daysBetween + 1;
     }
 
 
-    private double calculateCorrectnessModifier(Content content, OffsetDateTime today) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java]: Your logic was WAAY to complicated here. It's really way simpler. 😅

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java [hunk 22]
```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
...
     private int calculateDaysOverdue(Content content) {
         OffsetDateTime today = OffsetDateTime.now();
         OffsetDateTime repetitionDate = content.getUserProgressData().getNextLearnDate();
 
-        System.out.println("Content: " + content.getId());
-        System.out.println("Today: " + today);
-        System.out.println("Repetition Date: " + repetitionDate);
-
-        // Calculate the number of days between the current date and the repetition date
-        long daysBetween = ChronoUnit.DAYS.between(today, repetitionDate);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java]: Don't use ChronoUnit.DAYS.between because here it returns a negative number, which messed up the whole calculation.  Duration.between(today, repetitionDate).abs().toDays(); is the way to go (abs() makes it an absolute value)

**Review by PaulBredl — APPROVED** [review]
